codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java

```diff
@@ -336,28 +326,24 @@ public String mapShape(MapShape shape) {
   public String booleanShape(BooleanShape shape) {
     writer.addImportFromModule("github.com/dafny-lang/DafnyRuntimeGo", "dafny");
     var nilCheck = "";
+    String unAssertDataSource = dataSource.startsWith("input.(") ? "input" : dataSource;
     if (this.isOptional) {
       if (this.isPointable) {
-        nilCheck = "if %s == nil { return nil }".formatted(dataSource);
+        nilCheck = "if %s == nil { return nil }".formatted(unAssertDataSource);
       } else {
-        nilCheck = "if %s == nil { return b }".formatted(dataSource);
+        nilCheck = "if %s == nil { return b }".formatted(unAssertDataSource);
       }
     }
     return """
     func() %sbool {
         var b bool
         %s
-        b = %s%s
+        b = %s.(bool)
```

Comment:
Do we always require assertion for `bool` type?

Reply:
Yes it is required.